content/numerical/NumberTheoreticTransform.h

```diff
@@ -18,34 +18,31 @@ const ll mod = (119 << 23) + 1, root = 62; // = 998244353
 // and 483 << 21 (same root). The last two are > 10^9.
 typedef vector<ll> vl;
 void ntt(vl &a) {
-    int n = sz(a), L = 31 - __builtin_clz(n);
-    static vl rt(2, 1);
-    static int curL = mod / 2;
-    for (static int k = 2; k < n; k *= 2) {
-        rt.resize(n);
-        ll z[] = {1, modpow(root, curL /= 2)};
-        rep(i, k, 2 * k) rt[i] = rt[i / 2] * z[i & 1] % mod;
-    }
-    vl rev(n);
-    rep(i, 0, n) rev[i] = (rev[i / 2] | (i & 1) << L) / 2;
-    rep(i, 0, n) if (i < rev[i]) swap(a[i], a[rev[i]]);
-    for (int k = 1; k < n; k *= 2)
-        for (int i = 0; i < n; i += 2 * k)
-            rep(j, 0, k) {
-                ll z = rt[j + k] * a[i + j + k] % mod;
-                ll &ai = a[i + j];
-                a[i + j + k] = (z > ai ? ai - z + mod : ai - z);
-                ai += (ai + z >= mod ? z - mod : z);
-            }
+	int n = sz(a), L = 31 - __builtin_clz(n);
+	static vl rt(2, 1);
+	for (static int k = 2, l = 2; k < n; k *= 2, l++) {
+		rt.resize(n);
+		ll z[] = {1, modpow(root, mod >> l)};
```

Comment:
how terrible would `mod >> l++` be? also, let's avoid the name `l`, confusable with `1`

Reply:
not worth it imo, since it basically saves just 2 characters.